Assets/Tests/PlayModeTests/Scenarios/MV_ManualView.cs

```diff
@@ -0,0 +1,393 @@
+            (service, arg1, arg2) => service.StopViewWithID(arg1, arg2), // 10
+            (service, arg1, _) => service.PauseViewWithID(arg1), // 11
+            (service, arg1, _) => service.ResumeViewWithID(arg1), // 12
+            (service, _, arg2) => service.StopAllViews(arg2), // 13
+            (service, _, arg2) => service.SetGlobalViewSegmentation(arg2), // 14
+            (service, arg1, arg2) => service.AddSegmentationToViewWithID(arg1, arg2), // 15
+            (service, arg1, arg2) => service.AddSegmentationToViewWithName(arg1, arg2), // 16
+            (service, _, arg2) => service.UpdateGlobalViewSegmentation(arg2), // 17
+         };
+
+        private void AssertViewEventProperties(CountlyEventModel viewEvent, string expectedEventID, string expectedPreviousViewID, int? expectedDuration, int? expectedVisit, bool expectStart)
+        {
+            Assert.AreEqual(viewEvent.EventID, expectedEventID);
+            Assert.AreEqual(viewEvent.PreviousViewID, expectedPreviousViewID);
+
+            if (expectedDuration.HasValue) {
+                Assert.AreEqual(viewEvent.Duration, expectedDuration.Value);
+            }
+
+            if (expectStart) {
+                if (expectedVisit.HasValue) {
+                    Assert.AreEqual(viewEvent.Segmentation["visit"], expectedVisit.Value);
+                }
+            } else {
+                Assert.IsFalse(viewEvent.Segmentation.ContainsKey("visit"));
+                Assert.IsFalse(viewEvent.Segmentation.ContainsKey("start"));
+            }
+        }
+
+        Dictionary<string, object> testSegmentation;
+
+        // Initializes Countly and clears event repo and request repo
+        // Validates that repositories are clean before testing
+        [SetUp]
+        public void ScenarioTestSetup()
+        {
+            TestUtility.TestCleanup();
+
+            testSegmentation = TestUtility.TestSegmentation();
+            CountlyConfiguration config = new CountlyConfiguration("appKey", "serverURl")
+                .SetRequiresConsent(true);
+
+            config.GiveConsent(new Consents[] { Consents.Events, Consents.Views });
+            Countly.Instance.Init(config);
+
+            Countly.Instance.RequestHelper._requestRepo.Clear();
+            Countly.Instance.Views._eventService._eventRepo.Clear();
+
+            viewService = Countly.Instance.Views;
+            viewService.safeViewIDGenerator = new CustomIdProvider();
+
+            Assert.IsTrue(Countly.Instance.RequestHelper._requestRepo.Count == 0);
+            Assert.IsTrue(viewService._eventService._eventRepo.Count == 0);
+        }
+
+        // All public functions in ViewCountlyService
+        // Null values provided as viewID/viewName parameter for each function, with and without segmentation
+        // Nothing should crash and nothing should be recorded
+        [Test]
+        public void MV_100_badValues_null()
+        {
+            foreach (var func in allMethods) {
```

Comment:
hmm that is cool to collect all funcs to somewhere. But it reduces readability. I vote for it to stay but